pkg/stream/xprotocol/conn.go

```diff
@@ -153,6 +158,22 @@ func (sc *streamConn) Dispatch(buf types.IoBuffer) {
 		// 2.4 prepare next
 		sc.ctxManager.Next()
 	}
+
+	// do not need to check graceful shutdown since the remote server will close the connection.
+	if sc.isServerStream() {
+		sc.checkGracefulShutdown()
```

Comment:
There is no guarantee that we can close the connection when there are no new requests

Reply:
we also check it here: https://github.com/mosn/mosn/blob/766c14a8d91552d209093bd44436341ec2fd676d/pkg/stream/xprotocol/conn.go#L229

That means it will be closed when there are no new requests.

Reply:
okay, I got it.
Fixed it by checking it when deleting active stream in the proxy module.